syntax = "proto3";

message Location {
  int32 id = 1;
  int32 person_id = 2;
  string longitude = 3;
  string latitude = 4;
  string creation_time = 5;
}

message LocationList {
  repeated Location locations = 1;
}

message GetLocationRequest {
  int32 id = 1;
}

message GetLocationRangeRequest {
  int32 person_id = 1;
  string start_date = 2;
  string end_date = 3;
}

message CreateLocationRequest {
  int32 person_id = 1;
  string longitude = 2;
  string latitude = 3;
  string creation_time = 4;
}

message Empty {

}

service LocationService {
    rpc Get(Empty) returns (LocationList);
    rpc GetLocation (GetLocationRequest) returns (Location);
    rpc GetLocationRange (GetLocationRangeRequest) returns (LocationList);
    rpc Create(CreateLocationRequest) returns (Location);
}